modules/integration/tests-common/common-artifacts/lazy-loading-info/src/main/java/org/wso2/appserver/integration/common/artifacts/lazyloading/LazyLoadingInfoUtil.java

```diff
@@ -0,0 +1,176 @@
+                    webAppStatus.setWebAppStarted(true);
+                    log.info("Tenant " + tenantDomain + " Web-app: " + webAppName +
+                            " is available in configuration context.");
+                    boolean isWebAppGhost = Boolean.parseBoolean((String) webApplication.getProperty("GhostWebApp"));
+                    log.info("Tenant " + tenantDomain + " Web-app: " + webAppName + " is in Ghost deployment status :" +
+                            isWebAppGhost);
+                    webAppStatus.setWebAppGhost(isWebAppGhost);
+                } else {
+                    log.info("Given web-app:" + webAppName + " for tenant:" + tenantDomain + " not found in started state");
+                    webAppStatus.setWebAppStarted(false);
+                }
+
+            } else {
+                log.info("Tenant " + tenantDomain + " has no started web-apps.");
+                webAppStatus.setWebAppStarted(false);
+            }
+
+        } else {
+            log.info("Tenant " + tenantDomain + " configuration context is not loaded.");
+            webAppStatus.setTenantStatus(new TenantStatus(false));
+        }
+
+        return webAppStatus;
+
+    }
+
+
+    /**
+     * Check  the given web-app of super tenant is loaded.
+     *
+     * @param webAppName Web-app Name
+     * @return WebAppStatus  with current status  information about the Web app.
+     */
+
+    protected static WebAppStatus getSuperTenantWebAppStatus(String webAppName) {
+
+        WebAppStatus webAppStatus = new WebAppStatus();
+        webAppStatus.setTenantStatus(new TenantStatus(true)); // Super tenant always loaded;
+        ConfigurationContext serverConfigurationContext = getServerConfigurationContext();
+
+        WebApplicationsHolder webApplicationsHolder = (WebApplicationsHolder) ((HashMap)
+                serverConfigurationContext.getLocalProperty("carbon.webapps.holderlist")).get("webapps");
+        Map<String, WebApplication> startedWebAppMap = webApplicationsHolder.getStartedWebapps();
+        if (startedWebAppMap != null) {
+            WebApplication webApplication = startedWebAppMap.get(webAppName);
+            if (webApplication != null) {
+                webAppStatus.setWebAppStarted(true);
+                log.info("Super Tenant  Web-app: " + webAppName +
+                        " is available in configuration context.");
+                boolean isWebAppGhost = Boolean.parseBoolean((String) webApplication.getProperty("GhostWebApp"));
+                log.info("Super Tenant Web-app: " + webAppName + " is in Ghost deployment status :" +
+                        isWebAppGhost);
+                webAppStatus.setWebAppGhost(isWebAppGhost);
+            } else {
+                log.info("Given web-app:" + webAppName + " for super tenant  not found in started state");
+            }
+
+        } else {
+            log.info("Super Tenant has no started web-apps.");
+        }
+        return webAppStatus;
+    }
+
+
+}
```

Comment:
why do we need two methods here? Cant we get the status with previous method by passing "carbon.super"?